File: server/src/main/kotlin/com/blueoauld/server/domain/report/entity/type/ReportReason.kt
package com.blueoauld.server.domain.report.entity.type

enum class ReportReason(

    val label: String,
) {

    OBSCENITY("음란물"),
    MINOR("미성년자"),
    MONEY_TRANSACTION("금전거래"),
    ABUSE("욕설 및 협박"),
    IMPERSONATION("사칭 및 도용"),
    ETC("기타"),
}

File: ios/jipkok/Features/Report/Models/ReportReason.swift
enum ReportReason: CaseIterable, Hashable {

    case obscenity
    case minor
    case moneyTransaction
    case abuse
    case impersonation
    case etc

    var label: String {
        switch self {
        case .obscenity: "음란물"
        case .minor: "미성년자"
        case .moneyTransaction: "금전거래"
        case .abuse: "욕설 및 협박"
        case .impersonation: "사칭 및 도용"
        case .etc: "기타"
        }
    }
}
